panel-app/src/components/store.ts

```diff
@@ -185,23 +173,35 @@ function formatState() {
 
 	const formattedMetadata =
 		metadata &&
-		Object.keys(metadata).map((key) => {
+		Object.keys(metadata).reduce((result, key) => {
 			const { name, type, path, isBoolean } = metadata[key];
-			return {
-				[name]: { type, path, isBoolean },
-			};
-		});
+			result[name] = { type, path, isBoolean };
+			return result;
+		}, {});
 
 	const formattedSubItems =
 		subItems &&
 		subItems.map((item) => {
 			const { name, exclude, metadata } = item;
+			const formattedSubItemExclude =
+				exclude &&
+				exclude.map((item) => {
+					return { type: item.type, path: item.path };
+				});
+			const formattedSubItemMetadata =
+				metadata &&
+				Object.keys(metadata).reduce((result, key) => {
```

Comment:
I see this logic with `reduce` a few times in this file. Can we have a function `getFormattedMetadata()`?
They are all the same, no?
4/4